src/EventSender.cpp

```diff
@@ -347,6 +355,36 @@ void EventSender::update_nested_counters(QueryDesc *query_desc) {
   }
 }
 
+void EventSender::update_ignored_users(const char *new_guc_ignored_users) {
```

Comment:
IMO it's breaking abstractions: this function belongs to Config.cpp.
I'd call this function something like Config::sync() or something, cuz we likely would like to sync yagpcc.enable the same way (for another PR). I think it's okay to always call it and check for dirty hooks from there, so this mechanism also does not leak.
And both  ignored_users_set and ignored_users_guc_dirty may be made static to the Config module.

I don't think EventSender should be doing GUC parsing and refreshing by itself.